server/apps/immich/src/api-v1/asset/asset.service.ts

```diff
@@ -264,81 +247,18 @@ export class AssetService {
     res: Res,
     headers: Record<string, string>,
   ) {
-    const allowOriginalFile = !authUser.isPublicUser || authUser.isAllowDownload;
+    const allowOriginalFile = !!(!authUser.isPublicUser || authUser.isAllowDownload);
```

Comment:
What is the different between the old vs the new logic here?

Reply:
The old logic resulted in a type of `boolean | undefined`. That is OK because `undefined` also is falsy. However, it's not being passed into a function that  has a type `boolean`, so this (`!!`) is just a fancy way to turn `undefined` into `false`.

Reply:
No change in logic, just a coercion of type to align with the new method I added.